client-js/app/elements/labrad-registry.ts

```diff
@@ -61,31 +53,49 @@ export class LabradRegistry extends polymer.Base {
     return item.name.match(this.regex);
   }
 
-  selectKey() {
-    this.selDir = null;
-    this.selectType = 'key';
+  @listen('idx-changed')
+  changedSelected() {
+    console.log('ChangeSelected', this.idx);
   }
 
-  selectDir() {
-    this.selKey = null;
-    this.selectType = 'dir';
+  @computed()
+  selected(idx: number): boolean {
+    console.log("computing selected is: ", idx);
+    if (idx != null) {
+        return false;
+      } else {
+        console.log('doesnt exist');
+        return true;
+      }
+  }
+  @property({computed: '_computeSelectedType(idx)'})
+  selectedType: string;
```

Comment:
I tried that first, but the compiler doesn't like it. It seems selectedType as something odd like a function that returns a string, rather than just a string and it throws errors. This was the only way I could work it so the compiler would be happy.